networkit/cpp/components/ConnectedComponents.cpp

```diff
@@ -93,6 +93,9 @@ Graph ConnectedComponents::extractLargestConnectedComponent(const Graph &G, bool
 
     const auto compSizes = cc.getComponentSizes();
     if (compSizes.size() == 1) {
+        if (compactGraph) {
+            GraphTools::getCompactedGraph(G, GraphTools::getContinuousNodeIds(G));
```

Comment:
This doesn't return. Why is this not caught by tests?

Reply:
The new tests were failing but because of another reason. I fixed that now.